Source/WebKit/WebProcess/WebPage/WebFrame.cpp

```diff
@@ -256,14 +262,38 @@ void WebFrame::continueWillSubmitForm(FormSubmitListenerIdentifier listenerID)
 
 void WebFrame::didCommitLoadInAnotherProcess()
 {
-    // FIXME: Replace m_coreFrame with a RemoteFrame.
+    RefPtr coreFrame = m_coreFrame.get();
+    if (!coreFrame)
+        return;
+
+    RefPtr webPage = m_page.get();
+    if (!webPage)
+        return;
+
+    auto* corePage = webPage->corePage();
+    if (!corePage)
+        return;
+
+    RefPtr parent = coreFrame->tree().parent();
+    if (!parent)
+        return;
+
+    RefPtr ownerElement = coreFrame->ownerElement();
+    parent->tree().removeChild(*coreFrame);
+    coreFrame->disconnectOwnerElement();
+    auto client = makeUniqueRef<WebRemoteFrameClient>(*this);
+    auto newFrame = WebCore::RemoteFrame::create(*corePage, m_frameID, ownerElement.get(), WTFMove(client));
+    m_coreFrame = WeakPtr { newFrame.ptr() };
```

Comment:
Do we really need the explicit `WeakPtr { }`? Also, we should call `.get()` instead of `.ptr()` to avoid an unnecessary null check.

I'd be curious if this would build:
`m_coreFrame = newFrame;`